Mathlib/Order/ReltoGaloisConnection.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2024 Lagrange Mathematics and Computing Research Center. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anthony Bordg
+-/
+import Mathlib.Data.Rel
+import Mathlib.Order.GaloisConnection
+
+/-!
+# The Galois Connection Induced by a Relation
+
+In this file, we show that an arbitrary relation `R` between a pair of types `α` and `β` defines
+a pair `l` and `u` of adjoint order-preserving maps between the corresponding posets
+`Set α` and `(Set β)ᵒᵈ`.
+We define `lFixedPoints` (resp. `uFixedPoints`) as the set of fixed points `J` (resp. `I`) of
+`Set α` (resp. `(Set β)ᵒᵈ`) such that `u (l J) = J` (resp. `l (u I) = I`).
+
+## Main Results
+
+⋆ `to_galoisConnection`: we prove that the maps `l` and `u` form a Galois connection.
+⋆ `instEquivFixedPoints`: we prove that the said maps induce inverse bijections between the sets
+of fixed points.
+
+## References
+
+⋆ Engendrement de topologies, démontrabilité et opérations sur les sous-topos, Olivia Caramello and
+Laurent Lafforgue (in preparation)
+
+## Tags
+
+relation, Galois connection, induced bijection, fixed points
+-/
+
+section
+
+universe u v
+
+variable {α : Type u} {β : Type v} (R : Rel α β)
+
+namespace Rel
+
+/-! ### Pairs of adjoint maps defined by relations -/
+
+open OrderDual
+
+/-- `l` maps any set `J` of elements of type `α` to the set `{b : β | ∀ a ∈ J, R a b}` of
+elements `b` of type `β` such that `R a b` for every element `a` of `J`. -/
+def l (J : Set α) : (Set β)ᵒᵈ := toDual {b : β | ∀ a ∈ J, R a b}
```

Comment:
Could you please give this (and `u` below) a more descriptive name?

Reply:
@adamtopaz, I first used `lAdjMap` (resp.`rAdjMap`) for the left (resp. right) map of the adjunction, then realised it would be more coherent wrt to the file `GaloisConnection.lean` to use `l` and `u`.

Reply:
I think `l` should be named something like `rightDual` and `u` something like `leftDual`. Also, I think it's better to introduce these definitions without the order duals, develop an API for the definitions directly, then package that API in the Galois connection at the very end. So, for example, I envision that `rightDual` would be defined as 
```lean
def rightDual (J : Set α) : Set β := {b : β | ∀ a ∈ J, R a b}
```
and analogously for `leftDual`. 

Concerning the names, @YaelDillies would probably have a better sense of how to name these.

Reply:
> I think `l` should be named something like `rightDual` and `u` something like `leftDual`. 

I assume you mean the other way around, i.e. `l` should be named `leftDual`.

Reply:
Not sure I understand what API you envision, but I renamed `l` (resp. `u`) into `leftDual` (resp. `rightDual`).

Reply:
> I think it's better to introduce these definitions without the order duals, develop an API for the definitions directly, then package that API in the Galois connection at the very end.

I second this.

Reply:
@adamtopaz, Would be happy to implement this, but I would need a precise specification of what you want here.

Reply:
See https://github.com/leanprover-community/mathlib4/pull/14908#discussion_r1685417366